### PR TITLE
Update current route in willChangeRoute during an intermediate transition

### LINKED ISSUE #19449: [Bug] Router Service doesn't update currentRoute on loading states
### 🐞 Describe the Bug
The Router Service's `currentRoute` property doesn't update on loading states, even though `currentRouteName` does.

### 🔬 Minimal Reproduction
https://ember-twiddle.com/5dbab877e11a84a2ede3267d27f58c40

An example is provided in the Ember Twiddle. Note that the Twiddle is on Ember 3.18.1, but this also happens in the app I work on at Ember 3.24.1.

### 😕 Actual Behavior
When a route has a loading state and there are computeds/observers that rely on `router.currentRoute`, the `currentRoute` never shows loading state routes.

### 🤔 Expected Behavior
There should be loading routes in `currentRoute`.

### 🌍 Environment
- Ember: - 3.18.1, 3.24.1
- Node.js/npm: -
- OS: -
- Browser: -

### ➕ Additional Context
I found the same bug reported in the past at https://github.com/emberjs/ember.js/issues/17162, and it's closed now as seemingly fixed, but the bug is still present.

### PR DESCRIPTION
**Pull request by sly7-7**
Updated to follow @rwjblue comment.
Fixes #19449

### CODE REVIEW

File: packages/@ember/-internals/routing/lib/system/router.ts [hunk 1]
```diff
@@ -548,7 +548,7 @@ class EmberRouter extends EmberObject {
 
   intermediateTransitionTo(name: string, ...args: any[]) {
     this._routerMicrolib.intermediateTransitionTo(name, ...args);
-
+
```
Comment (rwjblue) [inline packages/@ember/-internals/routing/lib/system/router.ts:551]: some odd whitespace here

**Review by rwjblue — COMMENTED** [review]
We chatted a little bit about this in discord this afternoon, I think the thing we decided was:

* Update `router_js` to add an `isIntermediateTransition` to the `InternalTransition` base class
* Once landed + released, update EmberRouter's `routeWillChange` to detect that value and set `this.currentRoute` to that transitions `.to` value
* Add some inline comments explaining why 😛

**Comment by sly7-7** [comment]
Will need https://github.com/tildeio/router.js/pull/321 and a then router bump

**Review by rwjblue — APPROVED** [review]
Nice work! Thanks for picking this up!

**Pull request merged**